Support `increase/rate` function in the `MQE` query language. (#11591)
apache/skywalking: docs/en/api/metrics-query-expression.md

#### Result Type
The result type is determined by the type of selected not-null metric expression.

## Trend Operation
Trend Operation takes an expression and performs a trend calculation on its results.

Expression:
```text
<Trend-Operator>(Metrics Expression, time_range)
```

> `time_range` is the number of the calculates range, should be a positive integer. The unit will automatically adjusts according to the query [Step](../../../oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/enumeration/Step.java),

Reviewer: ```suggestion
`time_range` is the positive int of the calculated range. The unit will automatically align with to the query [Step](../../../oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/enumeration/Step.java),
```

for example, if the query Step is `MINUTE`, the unit of `time_range` is `minute`.

| Operator | Definition                                                                            | ExpressionResultType          |
|----------|---------------------------------------------------------------------------------------|-------------------------------|
| increase | returns the increase in the time range in the time series                             | TIME_SERIES_VALUES |
| rate     | returns the per-second average rate of increase in the time range in the time series  | TIME_SERIES_VALUES |

For example:
If we want to query the increase value of the `service_cpm` metric in 2 minute(assume the query Step is MINUTE),

> we can use the following expression:

Reviewer: Could you be more clear about this? If `increase in 2 minutes`, it should be T1 - 2, and T2 - 2.
The time sequence should be `T1-2`, `T2-2`, T1, T2.

I am not following your description. Am I getting this wrong?

Author: yes, the time sequence won't change, it's the metrics' time sequence.
Assume the step is minute,  the metrics result is `T1, T2, T3`. Then the `increase in 2 minutes` result is `V(T1)-V(T1-2), V(T2)-V(T2-2), V(T3)-V(T1)`.  
Is this description better？

Author: The description in the doc: `Assume the T3 point, the increase = T3 - T1, If the metric value(T3 or T1) is empty, the result will be empty.`
is described if the value of T1 or T3 is empty, in `Timepoint T3`, the increased value will be empty.

#### Result Type
The result type is determined by the type of selected not-null metric expression.

## Trend Operation
Trend Operation takes an expression and performs a trend calculation on its results.

Expression:
```text
<Trend-Operator>(Metrics Expression, time_range)
```

`time_range` is the positive int of the calculated range. The unit will automatically align with to the query [Step](../../../oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/enumeration/Step.java),
for example, if the query Step is `MINUTE`, the unit of `time_range` is `minute`.

| Operator | Definition                                                                            | ExpressionResultType          |
|----------|---------------------------------------------------------------------------------------|-------------------------------|
| increase | returns the increase in the time range in the time series                             | TIME_SERIES_VALUES |

> | rate     | returns the per-second average rate of increase in the time range in the time series  | TIME_SERIES_VALUES |

Reviewer: Should we consider supporting labeled value as well?
